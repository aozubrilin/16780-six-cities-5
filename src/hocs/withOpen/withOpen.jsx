import React from "react";

export const withOpen = (Component) => (props) => {
  const [isOpen, setOpen] = React.useState(false);

  const handleOpenChange = () => setOpen(!isOpen);

  return (
    <Component
      {...props}
      isOpen={isOpen}
      onOpenChange={handleOpenChange}
    />
  );
};

export default withOpen;
