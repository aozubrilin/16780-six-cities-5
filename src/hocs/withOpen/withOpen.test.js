
import React, {Fragment} from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withOpen from "./withOpen";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <Fragment>
      {children}
    </Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withOpen(MockComponent);

it(`withOpen renders correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped >
      <Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});

