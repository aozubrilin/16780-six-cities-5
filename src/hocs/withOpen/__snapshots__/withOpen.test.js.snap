// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpen renders correctly 1`] = `null`;
